Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/CoreDocumentSynchronizationConfig.swift

```diff
@@ -299,7 +307,10 @@ final class CoreDocumentSynchronization: Codable, Hashable {
     }
 
     internal static func filter(forNamespace namespace: MongoNamespace) -> Document {
-        return [CodingKeys.namespace.rawValue: try? BSONEncoder().encode(namespace)]
+        guard let namespaceDoc = try? BSONEncoder().encode(namespace) else {
+            return [CodingKeys.namespace.rawValue: BSONNull()]
+        }
+        return [CodingKeys.namespace.rawValue: namespaceDoc]
```

Comment:
[style nit] could we get rid of the guard block and simplify this to `try? BSONEncoder().encode(namespace) ?? BSONNull()`?